### PR TITLE
PIN 1799

### PR DESCRIPTION
**Pull request by TonioGela**
- Adapting to new tenant apis
- Adapting to new tenant apis

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 1]
```diff
@@ -429,12 +429,16 @@ final case class ProcessApiServiceImpl(
...
-      )
-      flattenServices = eservices.flatMap(service =>
-        convertToFlattenEservice(service, callerSubscribedEservices, organizationsDetails)
-      )
+      tenants                   <- Future.traverse(eservices.map(_.producerId))(tenantManagementService.getTenant)
+      selfcareIds               <- tenants.toList.traverse(_.selfcareId.toFuture(MissingSelfcareId))
+      eservicesAndSelfcareIds = eservices.fproduct(e => tenants.find(_.id == e.producerId).flatMap(_.selfcareId))
+      organizationsDetails <- partyManagementService.getBulkInstitutions(selfcareIds)
+      eservicesAndDescription = eservicesAndSelfcareIds.map { case (eservice, maybeSelfcareId) =>
+        (eservice, organizationsDetails.found.find(_.id == maybeSelfcareId).map(_.description).getOrElse("Unknown"))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: isn't this comparing `UUID` and `Option[String]`?

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 2]
```diff
@@ -631,17 +638,14 @@ final case class ProcessApiServiceImpl(
...
     eservice: client.model.EService,
     agreementSubscribedEservices: Seq[AgreementManagementDependency.Agreement],
-    institutionsDetails: BulkInstitutions
+    description: String
   ): Seq[FlatEService] = {
     val flatEServiceZero: FlatEService = FlatEService(
       id = eservice.id,
       producerId = eservice.producerId,
       name = eservice.name,
       // TODO "Unknown" is a temporary flag
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: this todo has no more meaning here

File: src/main/scala/it/pagopa/interop/catalogprocess/service/impl/PartyManagementServiceImpl.scala [hunk 3]
```diff
@@ -22,19 +23,19 @@ final case class PartyManagementServiceImpl(invoker: PartyManagementInvoker, api
...
-    identifiers: BulkPartiesSeed
-  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[BulkInstitutions] = for {
-    uid <- getUidFuture(contexts)
-    request = api.bulkInstitutions(identifiers)(uid)
-    result <- invoker
-      .invoke(request, s"Institutions retrieved for identifiers ${identifiers.partyIdentifiers.mkString(",")}")
-  } yield result
+    identifiers: List[String]
+  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[BulkInstitutions] = withUid { uid =>
+    val ids     = BulkPartiesSeed(identifiers.map(UUID.fromString))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/service/impl/PartyManagementServiceImpl.scala]: ⚠️ `UUID.fromString` can throw an exception.
Why not using `toFutureUUID` here?

File: src/test/scala/it/pagopa/interop/catalogprocess/authz/ProcessApiAuthzSpec.scala [hunk 4]
```diff
@@ -49,6 +49,7 @@ class ProcessApiAuthzSpec extends AnyWordSpecLike with BeforeAndAfterAll with Au
       fakeAttributeRegistryManagementService,
       fakeAgreementManagementService,
       fakeAuthorizationManagementService,
+      new FakeTenantManagementService,
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/catalogprocess/authz/ProcessApiAuthzSpec.scala]: let's use a consistent approach 🙏

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 5]
```diff
@@ -455,6 +453,17 @@ final case class ProcessApiServiceImpl(
...
 
+  def getGetEserviceAndSelfcareId(
+    eservice: client.model.EService
+  )(implicit contexts: Seq[(String, String)]): Future[(client.model.EService, UUID)] = eservice
+    .pure[Future]
+    .mproduct(eservice =>
+      tenantManagementService
+        .getTenant(eservice.producerId)
+        .flatMap(_.selfcareId.toFuture(MissingSelfcareId))
+        .flatMap(_.toFutureUUID)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: this is a `String` for us, we should not convert it to `UUID`.
If you need this on line 435, I suggest something more like
```
organizationsDetails.found.find(_.id.toString == selfcareId)
```

**Review by galales — CHANGES_REQUESTED** [review]

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**